src/sidebar/store/modules/test/real-time-updates-test.js

```diff
@@ -209,4 +224,70 @@ describe('sidebar/store/modules/real-time-updates', () => {
       assert.isTrue(store.hasPendingUpdatesOrDeletions());
     });
   });
+
+  describe('pendingMentionCount', () => {
+    it('counts new annos with mentions and existing annos with added mentions', () => {
+      fakeAllAnnotations.returns([
+        {
+          // This annotation does not yet have a mention to current user
+          id: 'existing_anno_1',
+          mentions: [],
+        },
+        {
+          // This annotation already has a mention to current user
+          id: 'existing_anno_2',
+          mentions: [{ userid: 'current_user_id' }],
+        },
+        {
+          // This annotation has a mention to other user
+          id: 'existing_anno_3',
+          mentions: [{ userid: 'someone_else' }],
+        },
+      ]);
+
+      store.receiveRealTimeUpdates({
+        updatedAnnotations: [
+          {
+            // New annotation with no mentions. Ignored
+            id: 'new_anno_1',
+            group: 'group-1',
+            mentions: [],
+          },
+          {
+            // New annotation with mentions to other user. Ignored
+            id: 'new_anno_2',
+            group: 'group-1',
+            mentions: [{ userid: 'someone_else' }],
+          },
+          {
+            // New annotation with mentions to current user. Counted
+            id: 'new_anno_3',
+            group: 'group-1',
+            mentions: [{ userid: 'current_user_id' }],
+          },
+          {
+            // Existing annotation with new mention to current user. Counted
+            id: 'existing_anno_1',
+            group: 'group-1',
+            mentions: [{ userid: 'current_user_id' }],
+          },
+          {
+            // Existing annotation which already had a mention to current user.
+            // Ignored
+            id: 'existing_anno_2',
+            group: 'group-1',
+            mentions: [{ userid: 'current_user_id' }],
+          },
+          {
+            // Existing annotation which now mentions another user. Ignored
+            id: 'existing_anno_3',
+            group: 'group-1',
+            mentions: [{ userid: 'someone_else_two' }],
+          },
+        ],
+      });
+
+      assert.equal(store.pendingMentionCount(), 2);
```

Comment:
There is only one test case here with data that relates to various cases. Debugging a failure here will be difficult because it could be caused by a variety of bugs. I'd suggest to perhaps split this up into one parametrized test.